lib/src/client_options.dart

```diff
@@ -69,19 +70,24 @@ class ClientOptions {
   /// Allows to enable receiving raw gateway event
   bool dispatchRawShardEvent;
 
+  /// The [RetryOptions] to use when a shard fails to connect to the gateway.
+  RetryOptions shardReconnectOptions;
+
   /// Makes a new `ClientOptions` object.
-  ClientOptions(
-      {this.allowedMentions,
-      this.shardCount,
-      this.messageCacheSize = 100,
-      this.largeThreshold = 50,
-      this.compressedGatewayPayloads = true,
-      this.guildSubscriptions = true,
-      this.initialPresence,
-      this.shutdownHook,
-      this.shutdownShardHook,
-      this.dispatchRawShardEvent = false,
-      this.shardIds});
+  ClientOptions({
+    this.allowedMentions,
+    this.shardCount,
+    this.messageCacheSize = 100,
+    this.largeThreshold = 50,
+    this.compressedGatewayPayloads = true,
+    this.guildSubscriptions = true,
+    this.initialPresence,
+    this.shutdownHook,
+    this.shutdownShardHook,
+    this.dispatchRawShardEvent = false,
+    this.shardIds,
+    this.shardReconnectOptions = const RetryOptions(maxAttempts: 50), // 50 attempts won't ever reasonably be reached
```

Comment:
Based on [this](https://github.com/google/dart-neats/blob/master/retry/lib/retry.dart#L107), it actually never exceeds 31 :)